acceptance-tests/build.gradle

```diff
@@ -68,44 +60,51 @@ def allTestTasks =
             jenkinsVersion.default || currentJava.isCompatibleWith(jenkinsVersion.requiredJavaVersion)
         }
         .collect { jenkinsVersion ->
-            def jenkinsWarLocation =
-                new File("${project.gradle.gradleUserHomeDir}/jenkins-cache/${jenkinsVersion.version}/jenkins.war")
+            def downloadJenkinsTask =
+                tasks.register("downloadJenkins${jenkinsVersion.label}", Download) {
+                    src jenkinsVersion.downloadUrl
+                    dest file("${project.gradle.gradleUserHomeDir}/jenkins-cache/${jenkinsVersion.version}/jenkins.war")
+                    onlyIfModified true
+                    tempAndMove true
+                }
 
             def testTask =
                 jenkinsVersion.default
                     ? tasks.named("test")
-                    : tasks.register("test_${jenkinsVersion.label}", Test)
+                    : tasks.register("test${jenkinsVersion.label}", Test)
 
             testTask.configure {
-                dependsOn copyTestDependencies
+                def jenkinsWar = downloadJenkinsTask
+                def gradlePlugin = configurations.gradlePlugin
+
+                inputs.files(jenkinsWar)
+                    .withPropertyName("jenkinsWar")
+                    .withNormalizer(ClasspathNormalizer)
+                // Gradle doesn't support hpi extension, therefore reproducible-archives is required as well
+                inputs.files(gradlePlugin)
+                    .withPropertyName("gradlePlugin")
+                    .withNormalizer(ClasspathNormalizer)
 
                 onlyIf {
                     // Do not run on Windows as written here: https://github.com/jenkinsci/acceptance-test-harness/blob/master/docs/EXTERNAL.md
                     !OperatingSystem.current().isWindows()
                 }
 
-                doFirst {
-                    download.run {
-                        src jenkinsVersion.downloadUrl
-                        dest jenkinsWarLocation
-                        onlyIfModified true
-                        tempAndMove true
-                    }
+                javaLauncher = javaToolchains.launcherFor {
+                    languageVersion = jenkinsVersion.javaVersion
                 }
 
                 // If executed on TeamCity, we need to set the Chromedriver path
                 if (gradle.ciTeamCityBuild) {
                     systemProperty 'webdriver.chrome.driver', System.getenv('HOME') + '/.gradle/webdriver/chromedriver/chromedriver'
                 }
```

Comment:
❌ My comment was to use a `CommandLineArgumentProvider` to pass in the system property to the test task. This way you can avoid that the test task has the absolute path to the Chrome driver as an input.

Reply:
Sorry, missed this one. Moved the logic to the provider: https://github.com/jenkinsci/gradle-plugin/pull/220/commits/0ebfe53f808e0d57e4d08bc87eb101fd879241de

Reply:
nit, but I would pass `gradle.ciTeamCityBuild` as a param to not hold on `Gradle` as a field in your provider